packages/deployments/contracts/contracts/core/connext/libraries/LibCrossDomainProperty.sol

```diff
@@ -9,152 +9,56 @@ library LibCrossDomainProperty {
 
-  // ============ Enums ============
-
-  /**
-   * Contains information so the properties can be type-checked properly
-   */
-  enum Types {
-    Invalid, // 0
-    DomainAndSender // 1
-  }
-
-  // ============ Structs ============
-
-  /**
-   * Struct containing the domain and an address of the caller of a function on that
-   * domain.
-   */
-  struct DomainAndSender {
-    uint32 domain;
-    address sender;
-  }
-
   // ============ Constants ============
 
-  uint256 private constant PROPERTY_LEN = 25; // 1 byte identifer + 4 bytes domain + 20 bytes address
-  // default value is the TypedMemView null view
-  bytes29 public constant EMPTY = hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
-  bytes public constant EMPTY_BYTES = hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
-
-  // ============ Modifiers ============
-
-  /**
-   * @notice Asserts a property is of type `_t`
-   * @param _view The stored property
-   * @param _t The expected type
-   */
-  modifier typeAssert(bytes29 _view, Types _t) {
-    _view.assertType(uint40(_t));
-    _;
-  }
+  // 32 bytes amount + 4 bytes domain + 20 bytes address = 56 bytes;
+  uint256 private constant PROPERTIES_LENGTH = 56;
 
   // ============ Internal Functions ============
 
   /**
-   * @notice Checks that view is a valid property length
-   * @param _view The bytes string
-   * @return TRUE if length is valid
-   */
-  function isValidPropertyLength(bytes29 _view) internal pure returns (bool) {
-    uint256 _len = _view.len();
-    return _len == PROPERTY_LEN;
-  }
-
-  /**
-   * @notice Checks that the property is of the specified type
-   * @param _type the type to check for
-   * @param _property The property
-   * @return True if the property is of the specified type
+   * @notice
+   * @param
+   * @param
+   * @param
+   * @param
+   * @return
    */
-  function isType(bytes29 _property, Types _type) internal pure returns (bool) {
-    return propertyType(_property) == uint8(_type);
+  function formatCalldataWithProperties(
+    uint256 _amount,
+    uint32 _originDomain,
+    address _originSender,
+    bytes memory callData
+  ) internal returns (bytes memory) {
+    // Convert data to bytes
+    bytes memory properties = abi.encodePacked(_amount, _originDomain, _originSender);
+    return bytes.concat(callData, properties);
   }
 
-  /**
-   * @notice Checks that the property is of type DomainAndSender
-   * @param _property The property
-   * @return True if the property is of type DomainAndSender
-   */
-  function isDomainAndSender(bytes29 _property) internal pure returns (bool) {
-    return isValidPropertyLength(_property) && isType(_property, Types.DomainAndSender);
+  function amount(bytes memory data) internal returns (uint256) {
+    // create view
+    bytes29 typed = data.ref(0);
+    // The data will be packed with the properties appended to the data
+    uint256 dataLen = typed.len() - PROPERTIES_LENGTH;
```

Comment:
perhaps do a safety check that `typed.len() >= PROPERTIES_LENGTH` and give an appropriate revert